recordset.go

```diff
@@ -197,14 +201,19 @@ func (rcs *Recordset) Results() <-chan *Result {
 // Close all streams from different nodes. A successful close return nil,
 // subsequent calls to the method will return ErrRecordsetClosed.
 func (rcs *Recordset) Close() error {
+	// check and close cancelled channel
+	// this will broadcast to all commands listening to the channel
+	select {
+	case <-rcs.cancelled:
+	default:
+		close(rcs.cancelled)
+	}
+
```

Comment:
I don't see the benefit of this block. cancelled channel is already being closed exactly once. What am I missing?

ps. Sorry for the late reply, I had replied a few days ago and I just found out that when you begin a review, you should also explicitly finish it.